sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/collectionOperations.scala

```diff
@@ -1882,3 +1882,98 @@ case class ArrayRepeat(left: Expression, right: Expression)
   }
 
 }
+
+/**
+ * Remove all elements that equal to element from the given array
+ */
+@ExpressionDescription(
+  usage = "_FUNC_(array, element) - Remove all elements that equal to element from array.",
+  examples = """
+    Examples:
+      > SELECT _FUNC_(array(1, 2, 3, null, 3), 3);
+       [1,2,null]
+  """, since = "2.4.0")
+case class ArrayRemove(left: Expression, right: Expression)
+  extends BinaryExpression with ImplicitCastInputTypes {
+
+  override def dataType: DataType = left.dataType
+
+  override def inputTypes: Seq[AbstractDataType] =
+    Seq(ArrayType, left.dataType.asInstanceOf[ArrayType].elementType)
+
+  lazy val elementType: DataType = left.dataType.asInstanceOf[ArrayType].elementType
+
+  override def nullSafeEval(arr: Any, value: Any): Any = {
+    val elementType = left.dataType.asInstanceOf[ArrayType].elementType
+    val data = arr.asInstanceOf[ArrayData].toArray[AnyRef](elementType).filter(_ != value)
```

Comment:
We shouldn't use `!=` directly, use `TypeUtils.getInterpretedOrdering()` instead. See #21361.

Reply:
@ueshin Fixed. Could you please review again? Thanks a lot!